sprezzatura/main/models.py

```diff
@@ -134,6 +142,23 @@ class EarTrainingElementPage(Page, MenuPageMixin):
         ('image', ImageChooserBlock())
     ])
 
+    def get_context(self, request, *args, **kwargs):
+        ctx = super().get_context(request, *args, **kwargs)
+        et_level_parent = self.get_parent().get_parent()
+        ctx['et_level_nav'] = [
+            {'page':  el, 'active': True if el == et_level_parent else False}
+            for el in et_level_parent.get_siblings()
+        ]
+        et_element_container = self.get_parent()
+        ctx['et_elements_nav'] = [
+            {
```

Comment:
This little bit of logic is a bit hard to read and repeated three times. Maybe a utility function could help here?